File: app/src/main/java/com/galeria/android/MediaItem.java
package com.galeria.android;

import android.net.Uri;

final class MediaItem {
    final long id;
    final Uri uri;
    final String name;
    final String mimeType;
    final long dateAdded;
    final long size;
    final String relativePath;
    final String albumKey;
    final String albumName;

    MediaItem(long id, Uri uri, String name, String mimeType, long dateAdded, long size, String relativePath, String albumKey, String albumName) {
        this.id = id;
        this.uri = uri;
        this.name = name == null ? "Sem nome" : name;
        this.mimeType = mimeType == null ? "" : mimeType;
        this.dateAdded = dateAdded;
        this.size = size;
        this.relativePath = relativePath == null ? "" : relativePath;
        this.albumKey = albumKey == null || albumKey.isEmpty() ? "root" : albumKey;
        this.albumName = albumName == null || albumName.isEmpty() ? "Galeria" : albumName;
    }

    boolean isVideo() {
        return mimeType.startsWith("video/");
    }

    boolean isImage() {
        return mimeType.startsWith("image/");
    }
}

File: app/src/main/kotlin/com/galeria/android/MediaItem.kt
package com.galeria.android

import android.net.Uri

class MediaItem(
    @JvmField val id: Long,
    @JvmField val uri: Uri,
    name: String?,
    mimeType: String?,
    @JvmField val dateAdded: Long,
    @JvmField val size: Long,
    relativePath: String?,
    albumKey: String?,
    albumName: String?,
    @JvmField val duration: Long = 0L
) {
    @JvmField val name: String = name ?: "Sem nome"
    @JvmField val mimeType: String = mimeType ?: ""
    @JvmField val relativePath: String = relativePath ?: ""
    @JvmField val albumKey: String = if (albumKey.isNullOrEmpty()) "root" else albumKey
    @JvmField val albumName: String = if (albumName.isNullOrEmpty()) "Galeria" else albumName

    fun isVideo(): Boolean = mimeType.startsWith("video/")

    fun isImage(): Boolean = mimeType.startsWith("image/")
}
